### ISSUE TITLE
Failure when using external (npm) modules

### ISSUE THREAD

**Issue by camin-mccluskey** [akamai/edgeworkers-examples#23]
Hi repo maintainers, 

Thanks for the examples. I am having a hard time integrating external modules from npm. I have followed the pattern shown in the [storelocator example](https://github.com/akamai/edgeworkers-examples/tree/master/storelocator) (i.e. using rollup). After a bit of trial and error I can deploy the code to the EdgeWorker, however when triggering the EdgeWorker on our staging network I get a `503: Service Unavailable` response. Unfortunately there isn't much information other than that.

What I have noticed is that when importing the modules and bundling them with rollup, the EdgeWorker will execute if the module **is not called**, so bundling the code is fine but It's having an issue calling the bundled code. I'm getting the same issue with both a commonJS module (https://github.com/auth0/node-jsonwebtoken) and with an ES6 module (https://github.com/uuidjs/uuid). I've tested them separately but I'm somewhat out of ideas - hopefully you can help 🙂

**Comment by sscian** [comment]
Camin, can you provide a bit more information to the email edgeworkers@akamai.com

Please include a url (curl) that would exercise the EdgeWorker...

**Comment by joshjohnson** [comment]
Hi @camin-mccluskey,

For troubleshooting purposes, can you add "akamai-x-ew-debug" to the Pragma header in your HTTP request.  (See https://learn.akamai.com/en-us/webhelp/edgeworkers/edgeworkers-user-guide/GUID-F888493F-6186-4400-89B4-0AEDF872DFC9.html)

Once that request header is in place, the Akamai platform will add some additional response headers that will be useful for debugging.

**Comment by camin-mccluskey** [comment]
Hi @sscian I will send that info in an email now 

Hi @joshjohnson I have added that header but unfortnately nothing comes back in the failure case I'm describing. Typically (i.e. when I dont run the external libs) I do see the event handlers being triggered

**Comment by camin-mccluskey** [comment]
Hi @sscian @joshjohnson it appears the issue lies with the use of the crypto library. Is this something that EdgeWorkers supports (as the browser does, for example)

**Comment by devrel-edgeworkers** [comment]
I've successfully used CryptoES, I guess it depends if it supports your crypto needs.

Currently there isn't the built-in support for the web.crypto API in EdgeWorkers.

Steve...


On Aug 13, 2020 9:25 PM, Camin McCluskey <notifications@github.com> wrote:

Hi @sscian<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sscian&d=DwMCaQ&c=96ZbZZcaMF4w0F4jpN6LZg&r=TdYdCJmPCrgiInYP-Bbe2SFDH_JhGNViqXfHKJ6oGX0&m=mGr4UXRioROBhmJY19hQDsRk252SXfU1SBpOEJ_Htu4&s=2c_8Mhj28srLt77aKmLonM3Sr5XU2iRCvwJZYKAJcdM&e=> @joshjohnson<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_joshjohnson&d=DwMCaQ&c=96ZbZZcaMF4w0F4jpN6LZg&r=TdYdCJmPCrgiInYP-Bbe2SFDH_JhGNViqXfHKJ6oGX0&m=mGr4UXRioROBhmJY19hQDsRk252SXfU1SBpOEJ_Htu4&s=udKMhfZPywuIqYtscGQSnn1NqTvHmzBtzG-Pb89hSTo&e=> it appears the issue lies with the use of the crypto library. Is this something that EdgeWorkers supports (as the browser does, for example)

**Comment by DawChihLiou** [comment]
Hi @camin-mccluskey @sscian @joshjohnson , I ran into similar problem when writing an edge worker with TypeScript and a few npm modules. I was able to ship the EW with a custom build process with Rollup. 

I created a PR to share the build process as an example. https://github.com/akamai/edgeworkers-examples/pull/35

The build process compiles TypeScript file, resolves third-party CommonJS and ES modules. It will bundle all the modules and output a `main.js`file at the end.

Hopefully this helps.
